apps/api-journeys/src/app/modules/block/form/validateFormCredentials.ts

```diff
@@ -0,0 +1,136 @@
+import { createClient } from '@formium/client'
+import { GraphQLError } from 'graphql'
+
+import { Block } from '.prisma/api-journeys-client'
+
+import { FormBlockUpdateInput } from '../../../__generated__/graphql'
+
+interface ValidateFormCredentialsProps {
+  input: FormBlockUpdateInput
+  block: Pick<Block, 'projectId' | 'formSlug' | 'apiToken'>
+}
+
+export async function validateFormCredentials({
+  input,
+  block
+}: ValidateFormCredentialsProps): Promise<FormBlockUpdateInput> {
+  const { apiToken, projectId, formSlug } = getCredentials({ input, block })
+
+  // throws error if trying to edit projectId or formSlug without apiToken
+  if (apiToken == null && (input.projectId != null || input.formSlug != null)) {
+    throw new GraphQLError('provide api token first', {
```

Comment:
can you change the error message to help the client understand they need to update the relevant field?

```suggestion
    throw new GraphQLError('provide apiToken first', {
```